crates/bevy_math/src/interpolation/lerp.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{interpolation::utils::*, Quat, Vec2, Vec3, Vec3A, Vec4};
+
+/// Defines how a particular type will be interpolated
+pub trait Lerp: Sized {
+    /// Lerp, `t` is unclamped
+    fn lerp_unclamped(a: &Self, b: &Self, t: f32) -> Self;
```

Comment:
I don't think `t` should be bound to an `f32`.
Imo this should be an associated type.

Reply:
t is the lerping factor that ranges from 0 to 1

Reply:
Yes but it's possible a user could want to use an `f64` or someother fixed point type for example.